### PR TITLE
Profiler Sync Timers

### PR DESCRIPTION
**Pull request by kngott**
## Summary

Attempt 2. Tested on dogora in all configurations (no profiler, BL & Tiny).

Not the best methodology (adds a BL || Tiny macro test), but seems to do the job.

Will test on Perlmutter as well.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Base/AMReX_BLProfiler.H [hunk 1]
```diff
@@ -475,7 +475,7 @@ inline std::string BLProfiler::CommStats::CFTToString(CommFuncType cft) {
 #define BL_PROFILE_INITPARAMS()
 #define BL_PROFILE_FINALIZE()
 
-#define BL_TINY_PROFILE_INITIALIZE()   amrex::TinyProfiler::Initialize()
+#define BL_TINY_PROFILE_INITIALIZE()   amrex::TinyProfiler::Initialize(); BLProfileSync::InitParams()
```
Comment (jmsexton03) [inline Src/Base/AMReX_BLProfiler.H]:
```suggestion
#define BL_TINY_PROFILE_INITIALIZE()   amrex::TinyProfiler::Initialize(); amrex::BLProfileSync::InitParams()
```

File: Src/Base/AMReX_FabArrayCommI.H [hunk 2]
```diff
@@ -168,7 +171,7 @@ FabArray<FAB>::FillBoundary_finish ()
 
     BL_PROFILE("FillBoundary_finish()");
 
-    if (!fbd) { n_filled = IntVect::TheZeroVector(); return; }
+    if (!fbd) { n_filled = IntVect::TheZeroVector(); BL_PROFILE_SYNC_STOP(); return; }
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_FabArrayCommI.H]: The only thing `BL_PROFILE_SYNC_STOP()` does is to decrement the counter.  So I think we could simply do it before this line and remove the one at the end of this function.

**Review by WeiqunZhang — CHANGES_REQUESTED** [review]
Don't you need to modify CMakeLists.txt?
```
diff --git a/Src/Base/CMakeLists.txt b/Src/Base/CMakeLists.txt
index ab86c6dfcc..935be24c8e 100644
--- a/Src/Base/CMakeLists.txt
+++ b/Src/Base/CMakeLists.txt
@@ -260,7 +260,7 @@ target_sources( amrex
 # Profiling
 #   this source file has zero symbols in default conditions, which creates
 #   ranlib warnings, e.g., on macOS
-if(AMReX_BASE_PROFILE OR AMReX_FORTRAN)
+if(AMReX_BASE_PROFILE OR OR AMReX_TINY_PROFILE OR AMReX_FORTRAN)
    target_sources( amrex
      PRIVATE
        AMReX_BLProfiler.cpp
```

**Comment by kngott** [comment]
Yup. Sure do! Was going to test it with WarpX on Perlmutter when I some cycles today, to track down and test the needed changes and get some fun numbers out at the same time. (PM was down last night). Looks like you may have found it, though. 😄

File: Src/Base/AMReX_FabArrayCommI.H (hunk 2 above)
Comment (kngott) [reply Src/Base/AMReX_FabArrayCommI.H]: Ah, good point! I'll make that tweak.

**Comment by kngott** [comment]
Tested all combinations on Perlmutter. They all work.

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**